### PR TITLE
Feature flag for provenance field in status

### PR DESCRIPTION
**Pull request by chuangw6**
# Changes

/kind feature 

(*This is [an additive change](https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md#additive-changes)*)

Introduced a dedicated feature flag of boolean type named "enable-provenance-in-status"
in feature-flags configmap to enable the provenance field in status. The `provenance` field 
was introducted to *run status in https://github.com/tektoncd/pipeline/pull/5580 to record authenticated metadata about how a 
software artifact was built i.e. the source where remote resource came from.

By default, this feature flag is false.

Signed-off-by: Chuang Wang <chuangw@google.com>

***Related PRs***:
- #5580 
- #5397 

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Added a new boolean feature flag named "enable-provenance-in-status" in feature-flags configmap to enable the provenance field in status to be populated. This field in status aims to record authenticated metadata about how a software artifact was built i.e. the source where remote resource came from.
```

### CODE REVIEW

**Comment by JeromeJu** [comment]
Wondering if this would be better to be part of #5580?

**Comment by chuangw6** [comment]
> Thanks Chuang! Wondering if this would be better to be part of #5580?

Good point. I didn't include in that PR for 2 reasons: 1/ tried to make that PR as smaller as possible 2/ wanted to add the feature flag when the code is actually trying to populate the field with data. But it turns out the PR doing that becomes large as well, so I just created this PR to add feature flag only to make review a bit easier for reviewers :D

**Comment by JeromeJu** [comment]
Sounds good. Also, LGTM.

> > Thanks Chuang! Wondering if this would be better to be part of #5580?
> 
> Good point. I didn't include in that PR for 2 reasons: 1/ tried to make that PR as smaller as possible 2/ wanted to add the feature flag when the code is actually trying to populate the field with data. But it turns out the PR doing that is quite large already as well, so I just created this PR to add feature flag only to make review a bit easier for reviewers :D

**Comment by chuangw6** [comment]
Hey @jerop @afrittoli @dibyom, 

I've added this dedicated feature flag for the provenance field in status that was recently introduced in #5580. PTAL. Thank you!!!

File: docs/install.md
Comment (jerop) [inline docs/install.md:449]: we should also list this feature among the alpha features with its individual feature flag - https://github.com/tektoncd/pipeline/blob/main/docs/install.md#alpha-features

**Comment by jerop** [comment]
we shouldn't release v0.41 with https://github.com/tektoncd/pipeline/pull/5580 without the feature flag which is added in this PR, so adding this PR to the mileston -- cc @tektoncd/core-maintainers

**Comment by dibyom** [comment]
#5580 just adds the field - I don't think its actually being used yet

**Comment by jerop** [comment]
> #5580 just adds the field - I don't think its actually being used yet

ah that's good to know, the conversation above btwn @chuangw6 and @JeromeJu made me think it was, in that case we don't need to have it in the milestone

File: docs/install.md
Comment (chuangw6) [reply docs/install.md:449]: Thanks @jerop ! Good idea. I've added this to the table. PTAL!

Comment (chuangw6) [reply docs/install.md:449]: I've also updated the paragraph above the alpha features table. PTAL. Thank you very much!

cc @dibyom

**Comment by chuangw6** [comment]
/hold

make sure this PR and #5397 are in the same release.

**Comment by chuangw6** [comment]
/unhold 

b/c https://github.com/tektoncd/pipeline/releases/tag/v0.41.0 was cut.

File: docs/pipelineruns.md
Comment (dibyom) [inline docs/pipelineruns.md:1389]: Nit: provenance is not a list

How about something like  -  Metadata about resources used in the PipelineRun such as the source from where a remote pipeline definition was fetched

**Review by dibyom — APPROVED** [review]
one minor thing about the docs, otherwise LGTM

File: docs/pipelineruns.md
Comment (chuangw6) [reply docs/pipelineruns.md]: SGTM. Updated. Thanks for the suggestion here. @dibyom

**Review by vdemeester — APPROVED** [review]

File: docs/pipelineruns.md
Comment (dibyom) [reply docs/pipelineruns.md]: Can we also update the descriptions for provenance in install.md and the feature flag comment itself to match?

Comment (chuangw6) [reply docs/pipelineruns.md]: Sounds good. Updated! Thanks @dibyom !

**Pull request merged**
